File: src/main/java/org/example/kalkulationsprogramm/domain/Schnittbilder.java
package org.example.kalkulationsprogramm.domain;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

@Entity
@Getter
@Setter
public class Schnittbilder
{
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;

    @Column(name = "bild_url_schnittbild", nullable = false, unique = true)
    private String bildUrlSchnittbild;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "schnitt_achse_id", nullable = false)
    private SchnittAchse schnittAchse;
}

File: src/main/kotlin/org/example/kalkulationsprogramm/domain/Schnittbilder.kt
package org.example.kalkulationsprogramm.domain

import jakarta.persistence.*

@Entity
open class Schnittbilder {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    open var id: Long = 0L

    @Column(nullable = false, unique = true)
    open var bildUrlSchnittbild: String? = null

    @Column(nullable = false, unique = true)
    open var form: String? = null

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    open var kategorie: Kategorie? = null

}